Allow connection URIs for Mongo

robdaemon opened the pull request:
Allows a connection URI string to be used for connecting to Mongo. When
using a connection UR you cannot provide the SSL certificate in a field on
settings page. This is a limitation of the underlying driver.

Resolves #5084

[ci mongo]

Commit:
Allow connection URIs for Mongo

Allows a connection URI string to be used for connecting to Mongo. When
using a connection UR you cannot provide the SSL certificate in a field on
settings page. This is a limitation of the underlying driver.

Resolves #5084

[ci mongo]

paulrosenzweig reviewed (approved):
Left a few small comments, but LGTM!

If we keep updating db forms like this, it's probably worth formalizing how we do different form versions and backwards compatibility. Just two examples is probably too early for that IMO though.

Commit:
Code review cleanup

Making the JS code a little cleaner per Paul's feedback

Commit:
Making conn-uri readonly on subsequent edits

This adds the ability to make a FormInputWidget readOnly. We use this as
a hint that you cannot edit the mongo connection URI after it is saved.

TODO: obscure the URI, and ignore any change to it upon save of an
existing connection

Commit:
Set the URI field to be a 'password' after save

This causes it to be obscured like a password is.

camsaul reviewed (approved):
Works for me

Commit:
Editing mongo conn-uri

The conn-uri becomes a password field after it's been saved to avoid
displaying a password in cleartext.
